packages/core/src/document.ts

```diff
@@ -151,36 +149,38 @@ class Document extends EventEmitter {
   }
 
   /**
-   * Takes a document containing only the genesis record and syncs the document to the state as
-   * of the specific commit requested.  Intentionally does not register the document so that it
-   * does not get notifications about newer commits, since we want a specific commit here.
-   * @param doc - Document containing only the genesis record
-   * @param tip - CID of the Tip record that we want to load the document at
-   * @param dispatcher
+   * Takes the most recent known-about version of a document and a specific commit and returns a new
+   * Document instance representing the same document but set to the state of the document at the
+   * requested commit.  If the requested commit is for a branch of history that conflicts with the
+   * known current version of the document, throws an error. Intentionally does not register the new
+   * document so that it does not get notifications about newer commits, since we want it tied to a
+   * specific commit.
+   * @param id - DocID of the document including the requested commit
+   * @param doc - Most current version of the document that we know about
+   * @private
    */
-  static async _syncDocumentToCommit<T extends Doctype> (
-      doc: Document,
-      tip: CID,
-      dispatcher: Dispatcher): Promise<Document> {
-    // TODO: Assert that doc contains only the genesis record
+  static async loadAtCommit<T extends Doctype> (
```

Comment:
Unfortunately we need this separate load path in Document exposed in order for the caching in the Ceramic instance to work properly

Reply:
Ok, that's unfortunate. Hopefully we the Document class itself is not fully needed in the future in the same way anyway.